### PR TITLE
Added aten::matmul operation.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: torch_xla/csrc/xla_lower_util.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      lhs_dimensions[i] = rhs_dimensions[i];
+    } else if (rhs_dimensions[i] == 1) {
+      rhs_dimensions[i] = lhs_dimensions[i];
+    } else {
+      XLA_ERROR() << "Unsupported DotBroadcast: " << lhs_shape << " vs. "
+                  << rhs_shape;
+    }
+  }
+
+  xla::XlaOp blhs = lhs;
```
Comment (asuhan) [inline torch_xla/csrc/xla_lower_util.cpp]: `broadcasted_lhs`, `broadcasted_rhs` is not so bad, the right hand side expressions go to the next line anyway.

File: torch_xla/csrc/xla_lower_util.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                                            const xla::Shape& to_shape) {
+  xla::int64 rank_delta = to_shape.rank() - op_shape.rank();
+  XLA_CHECK_GT(rank_delta, 0);
+
+  std::vector<xla::int64> reshape_sizes(to_shape.rank(), 1);
+  std::copy(op_shape.dimensions().begin(), op_shape.dimensions().end(),
+            reshape_sizes.begin() + rank_delta);
+  xla::XlaOp result = xla::Reshape(op, reshape_sizes);
+
+  std::vector<xla::int64> bcast_sizes(
```
Comment (asuhan) [inline torch_xla/csrc/xla_lower_util.cpp]: `broadcast_sizes`

File: torch_xla/csrc/xla_lower_util.cpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  if ((lhs_shape.rank() == 1 && rhs_shape.rank() == 1) ||
+      (lhs_shape.rank() == 2 && rhs_shape.rank() == 2) ||
+      (lhs_shape.rank() == 2 && rhs_shape.rank() == 1)) {
+    return xla::Dot(lhs, rhs);
+  }
+  if (lhs_shape.rank() == 1 && rhs_shape.rank() == 2) {
+    xla::XlaOp reshaped_lhs = xla::Reshape(lhs, {1, lhs_shape.dimensions(0)});
+    return xla::Reshape(xla::Dot(reshaped_lhs, rhs), {rhs_shape.dimensions(1)});
+  }
+  if (lhs_shape.rank() >= 1 && rhs_shape.rank() >= 1 &&
```
Comment (asuhan) [inline torch_xla/csrc/xla_lower_util.cpp]: I assume you've looked into `xla::BatchDot` and doesn't do what we need, correct?

File: torch_xla/csrc/xla_lower_util.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/xla_lower_util.cpp]: SG

File: torch_xla/csrc/xla_lower_util.cpp (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/csrc/xla_lower_util.cpp]: Sure

File: torch_xla/csrc/xla_lower_util.cpp (hunk 3 above)
Comment (dlibenzi) [reply torch_xla/csrc/xla_lower_util.cpp]: Yep, does not do all the reshape and broadcast required by PT.

**Pull request merged**